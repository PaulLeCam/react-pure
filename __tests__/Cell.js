import React from 'react';
import TestRenderer from 'react-test-renderer';

import Cell from '../src/Cell';

describe('Cell', () => {
  it('renders a div with class `pure-u-1` by default', () => {
    const renderer = TestRenderer.create(<Cell className="my-cell" />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('applies the class based on the `size` property', () => {
    const renderer = TestRenderer.create(<Cell size="1/2" />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-u-sm-*` class based on the `sm` property', () => {
    const renderer = TestRenderer.create(<Cell size="1/2" sm="1" />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-u-md-*` class based on the `md` property', () => {
    const renderer = TestRenderer.create(<Cell size="1/2" md="1" />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-u-lg-*` class based on the `lg` property', () => {
    const renderer = TestRenderer.create(<Cell size="1/2" lg="1/3" />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-u-xl-*` class based on the `xl` property', () => {
    const renderer = TestRenderer.create(<Cell size="1/2" lg="2/3" xl="1/4" />);
    expect(renderer.toJSON).toMatchSnapshot();
  });
});
