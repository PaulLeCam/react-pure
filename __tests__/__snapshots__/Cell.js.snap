// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell adds the \`pure-u-lg-*\` class based on the \`lg\` property 1`] = `[Function]`;

exports[`Cell adds the \`pure-u-md-*\` class based on the \`md\` property 1`] = `[Function]`;

exports[`Cell adds the \`pure-u-sm-*\` class based on the \`sm\` property 1`] = `[Function]`;

exports[`Cell adds the \`pure-u-xl-*\` class based on the \`xl\` property 1`] = `[Function]`;

exports[`Cell applies the class based on the \`size\` property 1`] = `[Function]`;

exports[`Cell renders a div with class \`pure-u-1\` by default 1`] = `[Function]`;
